packages/kbn-apm-synthtrace/src/lib/apm/client/apm_synthtrace_kibana_client.ts

```diff
@@ -53,23 +52,34 @@ export class ApmSynthtraceKibanaClient {
       return kibanaUrl;
     });
   }
-  async fetchLatestApmPackageVersion(currentKibanaVersion: string) {
-    const url = `https://epr-snapshot.elastic.co/search?package=apm&prerelease=true&all=true&kibana.version=${currentKibanaVersion}`;
-    const response = await fetch(url, { method: 'GET' });
-    const json = (await response.json()) as Array<{ version: string }>;
-    const packageVersions = (json ?? []).map((item) => item.version).sort(Semver.rcompare);
-    const validPackageVersions = packageVersions.filter((v) => Semver.valid(v));
-    const bestMatch = validPackageVersions[0];
-    if (!bestMatch) {
-      throw new Error(
-        `None of the available APM package versions matches the current Kibana version (${currentKibanaVersion}). The latest available version is ${packageVersions[0]}. This can happen if the Kibana version was recently bumped, and no matching APM package was released. Reach out to the fleet team if this persists.`
-      );
-    }
-    return bestMatch;
+  async fetchLatestApmPackageVersion(
+    kibanaUrl: string,
+    version: string,
+    username: string,
+    password: string
+  ) {
+    const url = `${kibanaUrl}/api/fleet/epm/packages/apm`;
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        Authorization: 'Basic ' + Buffer.from(username + ':' + password).toString('base64'),
```

Comment:
Nit: Headers can be wrapped in a function as they're also used in line 85

Reply:
Fixed in https://github.com/elastic/kibana/pull/141624/commits/313fc901030792f973eaf08859491b463fdcacd9